client/internal/peer/conn_status.go

```diff
@@ -12,7 +16,34 @@ const (
 )
 
 // ConnStatus describe the status of a peer's connection
-type ConnStatus int
+type ConnStatus int32
+
+// AtomicConnStatus is a thread-safe wrapper for ConnStatus
+type AtomicConnStatus struct {
+	status atomic.Int32
```

Comment:
I think we could remove the wrapper as it's only a single field, but that's a minor thing

Reply:
I did the wrapper because of the atom.Int32 stores Int32 and the ConnStatus is a custom type (because of the String method). So I tried to avoid the cast steps.

Reply:
`type AtomicConnStatus atomic.Int32` instead of the struct. But it's all right